cdf_fabric_replicator/time_series.py

```diff
@@ -75,13 +76,24 @@ def run(self) -> None:
                 time.sleep(sleep_time)
 
     def process_subscriptions(self) -> None:
-        for subscription in self.config.subscriptions:
-            for partition in subscription.partitions:
-                with ThreadPoolExecutor() as executor:
-                    future = executor.submit(
-                        self.process_partition, subscription, partition
-                    )
-                    logging.debug(future.result())
+        if (
```

Comment:
What about unit tests for this new functionality?

Reply:
Added unit tests